### PR TITLE
Add L0 qa test for perf_analyzer with response cache enabled

### PR DESCRIPTION
**Pull request by rmccorm4**
Depends on #3569

### CODE REVIEW

**Comment by tgerdesnv** [comment]
I'll defer my review to @jbkyang-nvi

File: qa/L0_perf_analyzer_report/models/identity_onnx_cache_off/1/model.onnx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+
+b
+y
+
+
+
+
+B
```
Comment (jbkyang-nvi) [inline qa/L0_perf_analyzer_report/models/identity_onnx_cache_off/1/model.onnx:15]: we usually generate models with `gen_qa_models`. Can we add this model and its configs there?

**Review by jbkyang-nvi — COMMENTED** [review]
looks good one comment though

**Comment by tanmayv25** [comment]
Is there a reason why we are not using any existing models? There are a wide-variety of models(both simple and ensemble) we have available on dlcluster which are generated as by gen_qa_models as described by katherine.

**Comment by rmccorm4** [comment]
@tanmayv25 @jbkyang-nvi updated test to use existing qa_model_repo models and tweak their model configs on the fly.

Example test output:
```
root@RMCCORMICK:/mnt/triton/jira/3252-cache-pa-test/server/qa/L0_perf_analyzer_report# ./test.sh 
=== Running /opt/tritonserver/bin/tritonserver --model-repository=/mnt/triton/jira/3252-cache-pa-test/server/qa/L0_perf_analyzer_report/models --response-cache-byte-size=8192
================================================================
[PERMUTATION] Protocol=http Model=simple_onnx_float32_float32_float32_cache_enabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 2526.4 infer/sec. Avg latency: 394 usec (std 84 usec)
  Pass [2] throughput: 2446.6 infer/sec. Avg latency: 408 usec (std 110 usec)
  Pass [3] throughput: 2447.6 infer/sec. Avg latency: 407 usec (std 98 usec)
  Client: 
    Request count: 12238
    Throughput: 2447.6 infer/sec
    Avg latency: 407 usec (standard deviation 98 usec)
    p50 latency: 377 usec
    p90 latency: 459 usec
    p95 latency: 572 usec
    p99 latency: 916 usec
    Avg HTTP time: 404 usec (send 67 usec + response wait 337 usec + receive 0 usec)
  Server: 
    Inference count: 14581
    Execution count: 14581
    Cache hit count: 0
    Successful request count: 14581
    Avg request latency: 129 usec (overhead 109 usec + queue 8 usec + cache hit/miss 12 usec)
      Average Cache Hit Latency: 12 usec
      Average Compute Latency (cache miss only): 0 usec 

  Composing models: 
  onnx_float32_float32_float32_cache_enabled, version: 
      Inference count: 0
      Execution count: 0
      Cache hit count: 14581
      Successful request count: 14581
      Avg request latency: 21 usec (overhead 1 usec + queue 8 usec + cache hit/miss 12 usec)
          Average Cache Hit Latency: 12 usec
          Average Compute Latency (cache miss only): 0 usec (compute input 0 usec + compute infer 0 usec + compute output 0 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 2447.6 infer/sec, latency 407 usec
================================================================
[PERMUTATION] Protocol=http Model=simple_onnx_float32_float32_float32_cache_disabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 772.6 infer/sec. Avg latency: 1292 usec (std 182 usec)
  Pass [2] throughput: 760.4 infer/sec. Avg latency: 1313 usec (std 163 usec)
  Pass [3] throughput: 774.8 infer/sec. Avg latency: 1288 usec (std 146 usec)
  Client: 
    Request count: 3874
    Throughput: 774.8 infer/sec
    Avg latency: 1288 usec (standard deviation 146 usec)
    p50 latency: 1294 usec
    p90 latency: 1436 usec
    p95 latency: 1481 usec
    p99 latency: 1574 usec
    Avg HTTP time: 1272 usec (send 111 usec + response wait 1161 usec + receive 0 usec)
  Server: 
    Inference count: 4672
    Execution count: 4672
    Successful request count: 4672
    Avg request latency: 713 usec (overhead 371 usec + queue 71 usec + compute 271 usec)

  Composing models: 
  onnx_float32_float32_float32_cache_disabled, version: 
      Inference count: 4671
      Execution count: 4671
      Successful request count: 4671
      Avg request latency: 581 usec (overhead 239 usec + queue 71 usec + compute input 83 usec + compute infer 132 usec + compute output 56 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 774.8 infer/sec, latency 1288 usec
================================================================
[PERMUTATION] Protocol=http Model=onnx_float32_float32_float32_cache_enabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 2788.2 infer/sec. Avg latency: 357 usec (std 90 usec)
  Pass [2] throughput: 2669 infer/sec. Avg latency: 373 usec (std 77 usec)
  Pass [3] throughput: 2557.8 infer/sec. Avg latency: 390 usec (std 98 usec)
  Client: 
    Request count: 12789
    Throughput: 2557.8 infer/sec
    Avg latency: 390 usec (standard deviation 98 usec)
    p50 latency: 354 usec
    p90 latency: 506 usec
    p95 latency: 574 usec
    p99 latency: 677 usec
    Avg HTTP time: 379 usec (send 67 usec + response wait 312 usec + receive 0 usec)
  Server: 
    Inference count: 0
    Execution count: 0
    Cache hit count: 15532
    Successful request count: 15532
    Avg request latency: 24 usec (overhead 3 usec + queue 12 usec + cache hit/miss 9 usec)
      Average Cache Hit Latency: 9 usec
      Average Compute Latency (cache miss only): 0 usec (compute input 0 usec + compute infer 0 usec + compute output 0 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 2557.8 infer/sec, latency 390 usec
================================================================
[PERMUTATION] Protocol=http Model=onnx_float32_float32_float32_cache_disabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 1077.4 infer/sec. Avg latency: 926 usec (std 164 usec)
  Pass [2] throughput: 1084.2 infer/sec. Avg latency: 920 usec (std 188 usec)
  Pass [3] throughput: 1103.6 infer/sec. Avg latency: 904 usec (std 137 usec)
  Client: 
    Request count: 5518
    Throughput: 1103.6 infer/sec
    Avg latency: 904 usec (standard deviation 137 usec)
    p50 latency: 880 usec
    p90 latency: 1109 usec
    p95 latency: 1131 usec
    p99 latency: 1169 usec
    Avg HTTP time: 885 usec (send 98 usec + response wait 787 usec + receive 0 usec)
  Server: 
    Inference count: 6695
    Execution count: 6695
    Successful request count: 6695
    Avg request latency: 360 usec (overhead 92 usec + queue 52 usec + compute input 66 usec + compute infer 116 usec + compute output 34 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 1103.6 infer/sec, latency 904 usec
================================================================
[PERMUTATION] Protocol=grpc Model=simple_onnx_float32_float32_float32_cache_enabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 1027.4 infer/sec. Avg latency: 971 usec (std 159 usec)
  Pass [2] throughput: 1059.8 infer/sec. Avg latency: 942 usec (std 139 usec)
  Pass [3] throughput: 1040.6 infer/sec. Avg latency: 959 usec (std 375 usec)
  Client: 
    Request count: 5203
    Throughput: 1040.6 infer/sec
    Avg latency: 959 usec (standard deviation 375 usec)
    p50 latency: 963 usec
    p90 latency: 1095 usec
    p95 latency: 1145 usec
    p99 latency: 1198 usec
    Avg gRPC time: 934 usec (marshal 5 usec + response wait 926 usec + unmarshal 3 usec)
  Server: 
    Inference count: 6274
    Execution count: 6274
    Cache hit count: 0
    Successful request count: 6274
    Avg request latency: 482 usec (overhead 421 usec + queue 24 usec + cache hit/miss 37 usec)
      Average Cache Hit Latency: 37 usec
      Average Compute Latency (cache miss only): 0 usec 

  Composing models: 
  onnx_float32_float32_float32_cache_enabled, version: 
      Inference count: 0
      Execution count: 0
      Cache hit count: 6274
      Successful request count: 6274
      Avg request latency: 65 usec (overhead 4 usec + queue 24 usec + cache hit/miss 37 usec)
          Average Cache Hit Latency: 37 usec
          Average Compute Latency (cache miss only): 0 usec (compute input 0 usec + compute infer 0 usec + compute output 0 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 1040.6 infer/sec, latency 959 usec
================================================================
[PERMUTATION] Protocol=grpc Model=simple_onnx_float32_float32_float32_cache_disabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 796 infer/sec. Avg latency: 1254 usec (std 139 usec)
  Pass [2] throughput: 807.2 infer/sec. Avg latency: 1237 usec (std 171 usec)
  Pass [3] throughput: 798 infer/sec. Avg latency: 1251 usec (std 178 usec)
  Client: 
    Request count: 3990
    Throughput: 798 infer/sec
    Avg latency: 1251 usec (standard deviation 178 usec)
    p50 latency: 1236 usec
    p90 latency: 1457 usec
    p95 latency: 1480 usec
    p99 latency: 1526 usec
    Avg gRPC time: 1236 usec (marshal 5 usec + response wait 1228 usec + unmarshal 3 usec)
  Server: 
    Inference count: 4762
    Execution count: 4762
    Successful request count: 4762
    Avg request latency: 805 usec (overhead 470 usec + queue 65 usec + compute 270 usec)

  Composing models: 
  onnx_float32_float32_float32_cache_disabled, version: 
      Inference count: 4762
      Execution count: 4762
      Successful request count: 4762
      Avg request latency: 661 usec (overhead 327 usec + queue 65 usec + compute input 84 usec + compute infer 130 usec + compute output 55 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 798 infer/sec, latency 1251 usec
================================================================
[PERMUTATION] Protocol=grpc Model=onnx_float32_float32_float32_cache_enabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 2011.6 infer/sec. Avg latency: 496 usec (std 65 usec)
  Pass [2] throughput: 2027 infer/sec. Avg latency: 492 usec (std 80 usec)
  Pass [3] throughput: 2029.6 infer/sec. Avg latency: 491 usec (std 199 usec)
  Client: 
    Request count: 10148
    Throughput: 2029.6 infer/sec
    Avg latency: 491 usec (standard deviation 199 usec)
    p50 latency: 476 usec
    p90 latency: 562 usec
    p95 latency: 609 usec
    p99 latency: 727 usec
    Avg gRPC time: 478 usec (marshal 3 usec + response wait 473 usec + unmarshal 2 usec)
  Server: 
    Inference count: 0
    Execution count: 0
    Cache hit count: 12217
    Successful request count: 12217
    Avg request latency: 44 usec (overhead 4 usec + queue 23 usec + cache hit/miss 17 usec)
      Average Cache Hit Latency: 17 usec
      Average Compute Latency (cache miss only): 0 usec (compute input 0 usec + compute infer 0 usec + compute output 0 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 2029.6 infer/sec, latency 491 usec
================================================================
[PERMUTATION] Protocol=grpc Model=onnx_float32_float32_float32_cache_disabled
================================================================
*** Measurement Settings ***
  Batch size: 1
  Using "time_windows" mode for stabilization
  Measurement window: 5000 msec
  Using synchronous calls for inference
  Stabilizing using average latency

Request concurrency: 1
  Pass [1] throughput: 1187.8 infer/sec. Avg latency: 840 usec (std 205 usec)
  Pass [2] throughput: 1139.6 infer/sec. Avg latency: 876 usec (std 185 usec)
  Pass [3] throughput: 1200 infer/sec. Avg latency: 832 usec (std 197 usec)
  Client: 
    Request count: 6000
    Throughput: 1200 infer/sec
    Avg latency: 832 usec (standard deviation 197 usec)
    p50 latency: 839 usec
    p90 latency: 1035 usec
    p95 latency: 1169 usec
    p99 latency: 1248 usec
    Avg gRPC time: 824 usec (marshal 4 usec + response wait 818 usec + unmarshal 2 usec)
  Server: 
    Inference count: 7115
    Execution count: 7115
    Successful request count: 7115
    Avg request latency: 422 usec (overhead 155 usec + queue 51 usec + compute input 71 usec + compute infer 115 usec + compute output 30 usec)

Inferences/Second vs. Client Average Batch Latency
Concurrency: 1, throughput: 1200 infer/sec, latency 832 usec

***
*** Test Passed
***
```

**Review by tanmayv25 — APPROVED** [review]

**Review by jbkyang-nvi — APPROVED** [review]

**Pull request merged**